inc/mailcollectorfeature.class.php

```diff
@@ -308,10 +308,11 @@ public static function updateMailCollectorOnAuthorizationCallback(
                         'login' => $authorization->fields['email'],
                     ]
                 );
+                Html::redirect($mailcollector->getLinkURL());
```

Comment:
We should try to redirect to `$mailcollector->getLinkURL()` even if `$success === false`, and if we do not have a valid mailcollector ID (this is not an expected usecase), we should redirect to the mail collector list. Indeed `Html::back()` will probably redirect user on the last page of the external provider OAuth authorization page, and is may result in a redirection loop.

```php
        $mailcollector = new MailCollector();
        $mailcollector_id = $params[$mailcollector->getForeignKeyField()] ?? null;
        if ($mailcollector_id !== null && $mailcollector->getFromDB($mailcollector_id)) {
            if ($success) {
                // Store authorized email into MailCollector
                $mailcollector->update(
                    [
                        'id'    => $mailcollector_id,
                        'login' => $authorization->fields['email'],
                    ]
                );
            }
            Html::redirect($mailcollector->getLinkURL());
        }

        Html::back($mailcollector->getSearchURL());
```

Reply:
I'm pretty sure getLinkURL will not work as expected if getFromDB fails; see last commit